drivers/retained_mem/CMakeLists.txt

```diff
@@ -3,7 +3,10 @@
 zephyr_syscall_header(${ZEPHYR_BASE}/include/zephyr/drivers/retained_mem.h)
 
 zephyr_library()
+
 zephyr_library_sources_ifdef(CONFIG_USERSPACE retained_mem_handlers.c)
 zephyr_library_sources_ifdef(CONFIG_RETAINED_MEM_NRF_GPREGRET retained_mem_nrf_gpregret.c)
 zephyr_library_sources_ifdef(CONFIG_RETAINED_MEM_ZEPHYR_RAM   retained_mem_zephyr_ram.c)
 zephyr_library_sources_ifdef(CONFIG_RETAINED_MEM_ZEPHYR_REG   retained_mem_zephyr_reg.c)
+
+add_subdirectory_ifdef(CONFIG_RETAINED_MEM_RV3028 retained_mem_rv3028)
```

Comment:
we do not use sub-folders for drivers

Reply:
I suggested using subfolders here as these drivers don't occupying just a single line of the `CMakeLists.txt`. The drivers include private headers like `${ZEPHYR_BASE}/drivers/rtc/rtc_rv3028` which should not be leaked to other RTC drivers, hence the:

```
zephyr_library()
zephyr_library_sources(retained_mem_rv3028.c)
zephyr_library_include_directories(${ZEPHYR_BASE}/drivers/rtc/rtc_rv3028)
```

and adding multiple lines to the "parent" CMakeLists.txt per driver is also going to be a mess quickly. It is harmless and IMO cleaner.

Reply:
We have drivers adding multiple lines to cmake files, would follow what we already have e.g. https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/flash/CMakeLists.txt

Reply:
Without subfolders, we cannot construct a new target using only the `zephyr_library_*` functions. As @bjarki-andreasen mentioned, the private headers of the rtc_rv3028 driver would be included in all retained_mem/eeprom drivers.

Reply:
```
if(SYMBOL_NAME)
  ...code here...
endif()
```

Reply:
I understand that Zephyr’s review model is not well supported by GitHub, making it hard to track follow-up changes. However, this was one of my previous solutions. The issue with this approach is that when the rv3028 retmem/eeprom driver is enabled, the rv3028 headers become accessible to any other retmem/eeprom drivers that are also enabled.

By having CMakeLists in a separate subfolder, we can construct a new library target that keeps the rv3028 headers more private.

Another solution would be to use plain CMake functions instead of Zephyr’s wrapper interface, but that’s outside the scope of this PR.

Reply:
> The issue with this approach is that when the rv3028 retmem/eeprom driver is enabled, the rv3028 headers become accessible to any other retmem/eeprom drivers that are also enabled.

That's not really an issue.

Reply:
Ok, maybe I took it too far :) I'll revert it back to a "flat" structure

Reply:
Put that one private file under a different name e.g. rtc_rv_3028_priv.h ? I don't see the issue. There are private files in subsystems in zephyr that you can't just include by doing `#include <system/priv_file.h>` but if someone really wants to include it they can do `#include <../../../system/priv_file.h>`

Reply:
Also mark the API as hidden using doxygen comments